Sources/DistributedActors/Cluster/ClusterControl.swift

```diff
@@ -155,4 +155,77 @@ public struct ClusterControl {
     public func down(member: Cluster.Member) {
         self.ref.tell(.command(.downCommandMember(member)))
     }
+
+    public func joined(node: UniqueNode, within: Duration) async throws {
+        try await self.awaitMemberStatus(is: .up, within: within, node: node)
+    }
+
+    public func ensureNodes(_ status: Cluster.MemberStatus, within: Duration, nodes: [UniqueNode]) async throws {
```

Comment:
Good API but some rewordings perhaps...? Since we're pulling it out from a testing API to a full API maybe we can make it a bit nicer.

```
func waitFor(nodes: Set<UniqueNode>, _ status: Cluster.MemberStatus, within: Duration) async throws
func waitFor(nodes: Set<UniqueNode>, atLeast status: Cluster.MemberStatus, within: Duration) async throws
```

we need the "exact" and "at least" versions I think. At least means "the nodes are at least `.joining`" etc.

When the exact API sees at least one node that is already PAST the expected/requested status we can throw; but the other one we can still allow and succeed, say, a node was up but became down already and with `atLeast: .up` that'd be okey.

Reply:
Ah sorry I see you did the atLeast as well already 👍  (was reading top down)